### ISSUE TITLE
runtime/syntax/gnuplot.vim: Wrong match type/delimiter for macro highlighting

### ISSUE THREAD

**Issue by Quoor8ee** [vim/vim#5969]
**Describe the bug**
gnuplot syntax highlighting has glitches when macros are used:
```
@MACRO
spurious highlighting words in macro color
# vim: syn=gnuplot
```

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Save the above code into a file and open it in vim
2. Anything up to the first blank (the `spurious` in the above example) is highlighted in the same colour as `@MACRO`, incorrectly skipping newlines.

**Expected behavior**
`spurious` should not be highlighted at all. For macro names, highlighting should only match `[a-zA-Z0-9_]*` on a single line, just like gnuplot does (see below).

**Environment (please complete the following information):**
 - VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Apr  4 2020 13:23:00), but probably at least all versions younger than 6 years (via git blame) affected.
 - OS: Archlinux, Debian Buster, Gentoo, FreeBSD 11.2 …
 - Terminal: all affected.

**Additional context**
Line 3486 of https://sourceforge.net/p/gnuplot/gnuplot-main/ci/master/tree/src/command.c specifies how gnuplot reads characters into macro names via the C function `isalnum()`, plus `_`.
`runtime/syntax/gnuplot.vim` wrongly uses `syn region` for macro names, so I propose this patch:
```
482c482
< syn region gnuplotMacro               start="@" end=" "
---
> syn match gnuplotMacro                "@[a-zA-Z0-9_]*"
```
to better match vim's syntax highlighting with the semantics and the actual behaviour of the program.

**Comment by chrisbra** [comment]
did you contact the maintainer of the file?

**Comment by Quoor8ee** [comment]
Good point, I just did!

**Comment by gdupras** [comment]
This was fixed recently, in commit 47e13953ffdbb9f163b901196dec8c2100b72edd.
